azerothcore/azerothcore-wotlk
# Fix Scholomance Gandling encounter

## Changes Proposed:
-  Corrected a possible typo in the shadow portal spell which caused summoned adds to be spawned outside of the Barov room.
-  Adjusted Risen Guardian to not give any reputation on kill according to https://youtu.be/vRFfCtpNpcY?t=122.


## Issues Addressed:
https://github.com/azerothcore/azerothcore-wotlk/issues/4520
- Closes 6. bottom middle room of darkmaster Gandling encounter isn't working correctly: the gate isn't an interactive gobject(can't even click it in gm mode like the others). No spawns in the room and the gate never opens, You can't escape it.




## SOURCE:

Risen Guardians should not award reputation 
Current Behavior no mobs spawned:
![2021-04-11 15-14-58](https://user-images.githubusercontent.com/10816895/114304385-854f1580-9adb-11eb-9718-27c3636e9b7d.gif)


## Tests Performed:

- Tested the spell when being teleported in the room there are no mobs to kill.Can't open the door.
- Tested in game on linux/docker setup from windows client.
- Tested with the changes. When teleported to Barov room there are adds now. When I kill them the door opens.
![2021-04-11 15-24-08](https://user-images.githubusercontent.com/10816895/114304465-fbec1300-9adb-11eb-8ac8-d879e37aae06.gif)


## How to Test the Changes:

Gandling test:
1. Go to Scholomance.
2. Trigger him to start the encounter by killing all mobs in the other rooms.
3. Wait to be teleported by Shadow Portal to Barov Family Vault.
4. Check to see if there are any mobs. There shouldn't be any
5. Re-Build AC. (If on docker run `./bin/acore-docker-build`)
6. Go to Scholomance.
7.  Trigger him to start the encounter by killing all mobs in the other rooms.
8. Wait to be teleported by Shadow Portal to Barov Family Vault.
9. Check to see if there are any mobs. There should be mobs now.
10. Check to see if when you kill the mobs the door opens.

Risen Guardian test:
1. Go to Scholomance.
2. Trigger him to start the encounter by killing all mobs in the other rooms.
3. Wait to be teleported by Shadow Portal.
4. Kill Risen Guardians.
5. Check to see if you are awarded reputation.
6. Apply the DB query from 5dd13f83a89ed6d9f7bc287cfd6dcb8524b7b294 commit.
7. Go to Scholomance.
8. Trigger him to start the encounter by killing all mobs in the other rooms.
9. Wait to be teleported by Shadow Portal.
10. Kill Risen Guardians.
11. Check to see if you are awarded reputation. You should no long gain any reputation.

## Known Issues and TODO List:

## Target Branch(es):







 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
